snapcraft/internal/repo/snaps.py

```diff
@@ -0,0 +1,196 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+import sys
+import urllib
+from subprocess import check_call, check_output, CalledProcessError
+
+import requests_unixsocket
+from requests import exceptions
+
+from . import errors
+
+
+logger = logging.getLogger(__name__)
+_SNAPD_URI_ROOT_TEMPLATE = 'http+unix://%2Frun%2Fsnapd.socket/v2/{}'
+
+
+class SnapPackage:
+
+    @classmethod
+    def is_valid_snap(cls, snap):
+        return cls(snap).is_valid()
```

Comment:
I consider this very weird. A class method that makes an instance. I haven't seen it before, is it an accepted pattern?

I would prefer the caller to make the instance. Or, a little less preferable, to make is_valid_snap a module function instead of a class method.

Reply:
This is a common pattern, yes